test/unit/source/vector_tile_source.test.js

```diff
@@ -52,21 +54,25 @@ test('VectorTileSource', (t) => {
 
         const source = createSource({ url: "/source.json" });
 
-        source.on('source.load', () => {
-            t.deepEqual(source.tiles, ["http://example.com/{z}/{x}/{y}.png"]);
-            t.deepEqual(source.minzoom, 1);
-            t.deepEqual(source.maxzoom, 10);
-            t.deepEqual(source.attribution, "Mapbox");
-            t.end();
+        source.on('data', (e) => {
+            if (e.sourceDataType === 'metadata') {
+                t.deepEqual(source.tiles, ["http://example.com/{z}/{x}/{y}.png"]);
+                t.deepEqual(source.minzoom, 1);
+                t.deepEqual(source.maxzoom, 10);
+                t.deepEqual(source.attribution, "Mapbox");
+                t.end();
+            }
         });
 
         window.server.respond();
     });
 
-    t.test('fires "data" event', (t) => {
+    t.test('fires event with metadata property', (t) => {
         window.server.respondWith('/source.json', JSON.stringify(require('../../fixtures/source')));
         const source = createSource({ url: "/source.json" });
-        source.on('data', t.end);
+        source.on('data', (e)=>{
+            if (e.sourceDataType === 'metadata') t.end();
+        });
```

Comment:
Should this test be checking for the `'update'` (or `'content'`) event, rather than `'metadata'`?